packages/cli/test/models/network/NetworkController.test.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+require('../../setup');
+
+import sinon from 'sinon';
+import NetworkController from '../../../src/models/network/NetworkController';
+import NetworkFile from '../../../src/models/files/NetworkFile';
+import ProjectFile from '../../../src/models/files/ProjectFile';
+import { AppProjectDeployer } from '../../../src/models/network/ProjectDeployer';
+
+contract('NetworkController', function() {
+  let projectFile, networkFile, controller;
+
+  beforeEach(() => {
+    sinon.createStubInstance(AppProjectDeployer);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('_getAllSolidityLibNames()', () => {
+    let controllerMock;
+    beforeEach(() => {
+      projectFile = new ProjectFile('test/mocks/mock-stdlib/zos.json');
+      networkFile = new NetworkFile(projectFile, 'test');
+      controller = new NetworkController('test', {}, networkFile);
+      controllerMock = sinon.mock(controller);
+    });
+
+    it('should order dependencies correctly', () => {
+      /**
+       *
+       *  A => D, E, F
+       *  B => C, E, F, G
+       *  C => D, G
+       *  D => []
+       *  E => D, Y
+       *  F => G
+       *  G => E
+       *  Y => []
+       *
+       *  Expected ordering:
+       *
+       *  [Y, D, E, G, F, A, C, B]
```

Comment:
```suggestion
       *  [D, Z, Y, E, G, F, A, C, B]
```

Same (note that Y now depends on Z)